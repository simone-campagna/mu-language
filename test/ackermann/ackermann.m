### Ackermann.s function:
###
###              - n+1                          if m==0
###             /
### Ack(m,n) := -- Ack(m-1, 1)                  if m>0, n==0
###             \
###              - Ack(m-1, Ack(m, n-1))        if m>0, n>0
###

        /^_RdR_LLCACA
     /R=/^RPL+
`fAq=/R=\^RPL+
        \^P_1CA

`H"ackermann"rS^CA
