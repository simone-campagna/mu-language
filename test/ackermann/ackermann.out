Ackermann(0,0)==1... ok
Ackermann(1,1)==2... ok
Ackermann(2,2)==3... ok
Ackermann(3,3)==4... ok
Ackermann(0,0)==2... ok
Ackermann(1,1)==3... ok
Ackermann(2,2)==4... ok
Ackermann(3,3)==5... ok
Ackermann(0,0)==3... ok
Ackermann(1,1)==5... ok
Ackermann(2,2)==7... ok
Ackermann(3,3)==9... ok
Ackermann(0,0)==5... ok
Ackermann(1,1)==13... ok
Ackermann(2,2)==29... ok
Ackermann(3,3)==61... ok

